File: android/app/src/main/java/com/dontworry/app/domain/search/QueryTokenizer.kt
package com.dontworry.app.domain.search

import com.dontworry.app.domain.text.TextNormalizer

object QueryTokenizer {
    fun tokenizeQuery(query: String): List<String> {
        val normalized = TextNormalizer.normalize(query)
        if (normalized.isBlank()) return emptyList()
        return normalized.split(" ").filter { it.isNotBlank() }
    }

    fun tokenizeForIndex(title: String, content: String?): List<String> {
        val normalized = TextNormalizer.normalize(title + " " + content.orEmpty())
        if (normalized.isBlank()) return emptyList()
        return normalized.split(" ").filter { it.isNotBlank() }
    }
}

File: ios/DontWorryiOS/Domain/QueryTokenizer.swift
import Foundation

enum QueryTokenizer {
    static func tokenizeQuery(_ query: String) -> [String] {
        let normalized = TextNormalizer.normalize(query)
        guard !normalized.isEmpty else { return [] }
        return normalized.split(separator: " ").map(String.init)
    }

    static func tokenizeForIndex(title: String, content: String?) -> [String] {
        let normalized = TextNormalizer.normalize(title + " " + (content ?? ""))
        guard !normalized.isEmpty else { return [] }
        return normalized.split(separator: " ").map(String.init)
    }
}
